### PR TITLE
community: Fix Baichuan Chat.

### PR DESCRIPTION
**Pull request by baichuan-assistant**
- **Description:** Baichuan Chat (with both Baichuan-Turbo and Baichuan-Turbo-192K models) has updated their APIs. There are breaking changes. For example, BAICHUAN_SECRET_KEY is removed in the latest API but is still required in Langchain. Baichuan's Langchain integration needs to be updated to the latest version. 
  - **Issue:** #15206
  - **Dependencies:** None,
  - **Twitter handle:** None

@hwchase17.

### CODE REVIEW

**Comment by baichuan-assistant** [comment]
@baskaryan

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 1]
```diff
@@ -102,15 +93,12 @@ def lc_serializable(self) -> bool:
     """Baichuan custom endpoints"""
     baichuan_api_key: Optional[SecretStr] = None
     """Baichuan API Key"""
-    baichuan_secret_key: Optional[SecretStr] = None
```
Comment (chyroc) [inline libs/community/langchain_community/chat_models/baichuan.py:105]: Other users of ChatBaichuan, if you input a secret key, could it result in issues with future compatibility updates? If that's the case, perhaps retaining the secret key and incorporating a warning message would be a good strategy to maintain the system's compatibility. This is just my preliminary suggestion.

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 2]
```diff
@@ -205,12 +187,9 @@ def _generate(
             return generate_from_stream(stream_iter)
 
         res = self._chat(messages, **kwargs)
-
+        if not res.status_code == 200:
```
Comment (chyroc) [inline libs/community/langchain_community/chat_models/baichuan.py:190]:
```suggestion
        if res.status_code != 200:
```

File: libs/community/langchain_community/chat_models/baichuan.py [hunk 3]
```diff
@@ -221,43 +200,47 @@ def _stream(
         **kwargs: Any,
     ) -> Iterator[ChatGenerationChunk]:
         res = self._chat(messages, **kwargs)
-
+        if not res.status_code == 200:
```
Comment (chyroc) [inline libs/community/langchain_community/chat_models/baichuan.py:203]:
```suggestion
        if res.status_code != 200:
```

**Comment by baichuan-assistant** [comment]
@baskaryan @hwchase17 Can someone take a look at this? It is affecting our users. Thanks!! =)

File: libs/community/tests/unit_tests/chat_models/test_baichuan.py [hunk 4]
```diff
@@ -85,65 +84,34 @@ def test__convert_delta_to_message_human() -> None:
...
-    assert captured.out == "**********"
-
 
 def test_baichuan_key_masked_when_passed_via_constructor(
     capsys: CaptureFixture,
 ) -> None:
     """Test initialization with an API key provided via the initializer"""
     chat = ChatBaichuan(
-        baichuan_api_key="test-api-key", baichuan_secret_key="test-secret-key"
+        baichuan_api_key="test-api-key"
```
Comment (chyroc) [inline libs/community/tests/unit_tests/chat_models/test_baichuan.py:104]: My suggestion is not to remove this parameter to test compatibility

File: libs/community/langchain_community/chat_models/baichuan.py (hunk 1 above)
Comment (baichuan-assistant) [reply libs/community/langchain_community/chat_models/baichuan.py:105]: Adding secret key as input would not cause compatibility issues.

However I agree it is a good practice keep the old field. It is also easier for old users to understand what happened.

Updated.

File: libs/community/langchain_community/chat_models/baichuan.py (hunk 2 above)
Comment (baichuan-assistant) [reply libs/community/langchain_community/chat_models/baichuan.py:190]: Done.

File: libs/community/langchain_community/chat_models/baichuan.py (hunk 3 above)
Comment (baichuan-assistant) [reply libs/community/langchain_community/chat_models/baichuan.py:203]: Done.

File: libs/community/tests/unit_tests/chat_models/test_baichuan.py (hunk 4 above)
Comment (baichuan-assistant) [reply libs/community/tests/unit_tests/chat_models/test_baichuan.py:104]: Sure.

**Review by hwchase17 — APPROVED** [review]
i cannot test this code since I don't have access, but since it seems like you are affiliated wtih baichuan i will trust you

**Comment by baichuan-assistant** [comment]
> i cannot test this code since I don't have access, but since it seems like you are affiliated wtih baichuan i will trust you

Thanks! I have fixed the lint. PTAL :)

**Comment by baichuan-assistant** [comment]
@chyroc Fix doc lint. PTAL :)

**Review by chyroc — APPROVED** [review]

**Comment by baichuan-assistant** [comment]
@hwchase17 @chyroc Anything else I need to do here? It looks to me all checks have passed :)

**Comment by baichuan-assistant** [comment]
@hwchase17 @chyroc friendly ping :)

**Comment by baichuan-assistant** [comment]
+CC @efriis

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**